### PR TITLE
PUBDEV-8619:  MaxR runs very slow

### PR DESCRIPTION
**Pull request by wendycwong**
This PR tries to fix the problem in JIRA: https://h2oai.atlassian.net/browse/PUBDEV-8619.

I have added two fixes: 
1.  Minor fix to make sure replacement is only run for maxr when predictor subset size >= 2.
2. Strict check to make sure we don't built duplicate models.  I used an array structure as a unit to perform the duplication check.  In my latest code, I switched to using BitSet and the effect on run time is very obvious.  Refer to black curve in the plot I have attached here.

3.  If user does not specifiy regularization, I will turn it off because for maxR, we are considering small number of predictors anyway.  If users set the regularization, it will not be turned off but the code will run slower in this case.  

4.  I have incorporated Michalk's comment and removed the GLM model once it is deemed inferior.  This will clear up memory usage.

### CODE REVIEW

File: h2o-algos/src/main/java/hex/modelselection/ModelSelection.java [hunk 1]
```diff
@@ -227,7 +227,7 @@ void buildMaxRModels(ModelSelectionModel model) {
                         _parms, _foldColumn, _glmNFolds, _foldAssignment); // forward step
                 validSubset.removeAll(currSubsetIndices);
                 _job.update(predNum, "Finished building all models with "+predNum+" predictors.");
-                if (predNum < _numPredictors) {
+                if (predNum < _numPredictors && predNum > 1) {
                     GLMModel currBestR2Model = replacement(currSubsetIndices, coefNames, bestR2Model.r2(), _parms,
```
Comment (wendycwong) [inline h2o-algos/src/main/java/hex/modelselection/ModelSelection.java:231]: When you have just one predictor, there is no need to switch it out and no need to call replacement step.

**Comment by wendycwong** [comment]
Here is a comparison of run times from different versions of my fix.  The one in black is using code in this PR:

image

**Comment by wendycwong** [comment]
Latest run with latest change.  Note that I have to increase the memory requirement for no L1/L2 reg no duplication with BitSet run.

image

File: scripts/jenkins/Makefile.jenkins [hunk 2]
```diff
@@ -122,7 +122,7 @@ test-pyunit-fault-tolerance: init-lookup lookup-fault-tolerance-tests
 	export NO_GCE_CHECK=True && cd h2o-py/tests/ && ../../scripts/run.py --wipeall --geterrs --numclouds 1 --numnodes 3 --jvm.xmx 4g --testlist .lookup.txt
 
 test-pyunit-medium-large:
-	export NO_GCE_CHECK=True && cd h2o-py/tests/ && ../../scripts/run.py --wipeall --geterrs --testsize ml --numclouds 1 --numnodes 5 --jvm.xmx 15g
+	export NO_GCE_CHECK=True && cd h2o-py/tests/ && ../../scripts/run.py --wipeall --geterrs --testsize ml --numclouds 1 --numnodes 5 --jvm.xmx 20g
```
Comment (michalkurka) [inline scripts/jenkins/Makefile.jenkins:125]: @wendycwong do we still need this after the fix that removes the intermediate glm models from memory?

Comment (wendycwong) [reply scripts/jenkins/Makefile.jenkins:125]: Probably not.  Let me remove it.

**Review by michalkurka — APPROVED** [review]
Thank you @wendycwong!

**Pull request merged**